mtgjson5/providers/tcgplayer.py

```diff
@@ -147,6 +147,123 @@ def generate_today_price_dict(
 
         return dict(results)
 
+    def generate_today_buylist_price_dict(
+        self, all_printings_path: pathlib.Path
+    ) -> Dict[str, MtgjsonPricesObject]:
+        """
+        Download the TCGPlayer pricing API and collate into MTGJSON format
+        :param all_printings_path Path to AllPrintings.json for pre-processing
+        :return: Buylist Prices to combine with others
+        """
+
+        if not self.__keys_found:
+            LOGGER.warning("Keys not found for TCGPlayer, skipping")
+            return {}
+        uuid_map = generate_tcgplayer_to_mtgjson_map(all_printings_path)
+        CACHE_PATH.mkdir(parents=True, exist_ok=True)
+        with CACHE_PATH.joinpath("tcgplayer_buylist_price_map.json").open("w") as file:
+            json.dump(uuid_map, file)
+
+        ids_and_names = self.get_tcgplayer_magic_set_ids()
+
+        results = parallel_call(
+            get_tcgplayer_buylist_prices_map, ids_and_names, fold_dict=True
+        )
+
+        return dict(results)
+
+    def generate_buylist_price_sequential(
+        self, all_printings_path: pathlib.Path
+    ) -> List[Dict[str, MtgjsonPricesObject]]:
+        """
+        Download the TCGPlayer pricing API and collate into MTGJSON format using a sequential process
+        :param all_printings_path Path to AllPrintings.json for pre-processing
+        :return: Buylist Prices to combine, not in correct format yet
+        """
+        if not self.__keys_found:
+            LOGGER.warning("Keys not found for TCGPlayer, skipping")
+            return [{}]
+        uuid_map = generate_tcgplayer_to_mtgjson_map(all_printings_path)
+        CACHE_PATH.mkdir(parents=True, exist_ok=True)
+        with CACHE_PATH.joinpath("tcgplayer_buylist_price_map.json").open("w") as file:
+            json.dump(uuid_map, file)
+
+        ids_and_names = self.get_tcgplayer_magic_set_ids()
+        set_price_data = []
+        for set_ids in ids_and_names:
```

Comment:
Could use list comp here.
set_price_data = [get_tcg(set_ids) for set_ids in ids_and_names]

Reply:
Also your variable names don't match your function name. "ids_and_names" when the function says get_set_ids only...?